### ISSUE TITLE
Translation between package syntax and equation in paper

### ISSUE THREAD

**Issue by jmshapir** [JMSLab/xtevent#13]
If we type

`xtevent y_base x_r, pol(z) panelvar(id) timevar(t) pre(G) post(M) overidpre(L_G) overidpost(L_M) cluster(id) norm(-1)`

then we do not get a plot that matches equation (2) of [EventStudy](https://www.brown.edu/Research/Shapiro/pdfs/EventStudy.pdf#page=4).

Example below. 

Looks like we may want to adjust the package so that it interprets the inputs in a way that lines up with the equation in the paper.

```
use analysis/release/exampleplots/simulation_data_dynamic.dta
xtevent y_base x_r, pol(z) panelvar(id) timevar(t) pre(0) post(5) overidpre(5) overidpost(1) cluster(id) norm(-1)
xteventplot
```
![152226091-a78c9d4a-8565-472b-8c2d-c6bfa0587030](https://user-images.githubusercontent.com/7840353/152359622-282996d1-5e20-4a72-aa59-3ea34042eb82.png)

**Comment by veli-m-andirin** [comment]
It looks the solution might be as simple as adding -1 to the RHS [here](https://github.com/JMSLab/xtevent/blob/c01ec79fadc3814700a1fb9843efd87cd548a23f/xtevent/xtevent.ado#L167). I am not sure if this would break something else (it looks like it wouldn't, but just wanted to flag this possibility).

**Comment by jorpppp** [comment]
@Constantino-Carreto-Romero will:

- [ ] Check the equation in the paper
- [ ] Check how the G, L_G, M and L_M variables in the paper map to inputs in the command
- [ ] Check if the solution by @veli-m-andirin here: https://github.com/JMSLab/xtevent/issues/13#issuecomment-1043288029 works and if it may break something else.

**Comment by Constantino-Carreto-Romero** [comment]
Thanks @veli-m-andirin for pointing out that solution, I checked it out and it works perfectly. 
On the other hand, it turns out that when using `window` instead of the combined use of `pre, post, overidpre`, and `overidpost` happens the same issue. Applying the same solution to `window` will cause graphs not to be symmetrical. Therefore, we will change the definition of `window` and its equivalence with the combined use of `pre, post, overidpre`, and `overidpost` to continue producing symmetrical graphs. The only change will be in the helpfile.

**Comment by jmshapir** [comment]
Thanks @Constantino-Carreto-Romero!

Just confirming that, with the solution you have in mind, we'll have a mapping of this form:
* G-->pre
* M-->post
* L_G-->overidpre
* L_M-->overidpost

?

Per https://github.com/JMSLab/xtevent/issues/13#issue-1123137546, that sort of mapping seems helpful for avoiding confusion on the part of a user who is familiar with the equation in the paper. Thanks.

**Comment by Constantino-Carreto-Romero** [comment]
That's right, I'm following that mapping @jmshapir

**Comment by jorpppp** [comment]
Thanks @Constantino-Carreto-Romero, let me rewrite this helpfile a bit before we start a PR for this.

**Comment by jorpppp** [comment]
We checked the helpfile and this is ready to start a PR @Constantino-Carreto-Romero

**Comment by Constantino-Carreto-Romero** [comment]
Thanks you @jorpppp. 
While this issue was opened, we had a PR for another issue. Therefore, to avoid file conflicts when doing the merge, I have taken the latest versions of these files and added the changes from this issue. Now, I'm going to start the PR and close this issue.

**Comment by Constantino-Carreto-Romero** [comment]
Summary: in this issue, we adjusted the package's translation of the equation in the paper. We modified the equivalence between `window` and `pre, post, overidpre` and `overidpost` as well.
Thread continues in PR #31

**Comment by veli-m-andirin** [comment]
@Constantino-Carreto-Romero @jorpppp,

When I run the code snippet in https://github.com/JMSLab/xtevent/issues/13#issue-1123137546 I still see a plot where we have an extra lag. I think the furthest lag should be 6 if the mapping between the code and the paper is correct. Please let me know if I am missing something.

@nateschor @rcalvo12 fyi.

**Comment by nateschor** [comment]
@Constantino-Carreto-Romero @jorpppp When I run the code snippet in https://github.com/JMSLab/eventstudyr/pull/10#issuecomment-1175448366 I get the furthest lag as overidpost + post, but running the code in https://github.com/JMSLab/eventstudyr/pull/10#issuecomment-1175132904 gives the furthest lag as overidpost + post + 1, so it seems like whether or not the +1 is included depends upon the values of post and overidpost. Please let me know if I'm missing something.

@veli-m-andirin @rcalvo12 fyi

**Comment by nateschor** [comment]
@jmshapir sure thing!

@Constantino-Carreto-Romero @jorpppp following up on https://github.com/JMSLab/xtevent/issues/13#issuecomment-1176260606:

I ran `xtevent y_base x_r, pol(z) panel(id) timevar(t) pre(2) post(3) overidpre(4) overidpost(4) norm(-3) plot` and was expecting post + overidpost + 1 = 8 lags but am getting 7 now. I used `simulation_data_dynamic.dta` from the [paper archive](https://data.nber.org/data-appendix/w29170/)

Thank you!

![image](https://user-images.githubusercontent.com/70981003/177406194-12dd7c3e-7a5a-48d4-846f-e4fbf991590e.png)

**Comment by jmshapir** [comment]
Thanks @nateschor! Can you also post what you get when you execute the snippet in https://github.com/JMSLab/xtevent/issues/13#issue-1123137546?

**Comment by jmshapir** [comment]
@nateschor regarding [this example](https://github.com/JMSLab/xtevent/issues/13#issuecomment-1176284955), thinking of the translation between the [paper](https://scholar.harvard.edu/files/shapiro/files/eventstudy.pdf#page=5) and the package as roughly

`xtevent y_base x_r, pol(z) panelvar(id) timevar(t) pre(G) post(M) overidpre(L_G) overidpost(L_M) cluster(id) norm(k*)`

it seems like we'd expect the x-axis to extend to 

`M+L_M = post+overidpost`

which is what I see in the output you posted? (There, `post+overidpost=7` and the x-axis indeed extends to 7+.)

Can you let me know where I'm going wrong? Thanks!

**Comment by Constantino-Carreto-Romero** [comment]
@veli-m-andirin  @nateschor thanks for the follow-up.
xtevent 1.0.0 still presents the problem of adding one extra lag to the right window, but in xtevent 2.0.0 this problem was fixed. 
version 2.0.0 is already available to install by SSC. 
We are announcing changes from one version to the other in [the repository](https://github.com/JMSLab/xtevent/tree/issue58_installation_from_github), but still working on it.
Let me know if the problem persists. Thanks.

**Comment by nateschor** [comment]
> Thanks @nateschor! Can you also post what you get when you execute the snippet in https://github.com/JMSLab/xtevent/issues/13#issue-1123137546?

@jmshapir when I execute the code in https://github.com/JMSLab/xtevent/issues/13#issue-1123137546 I get the plot below, which matches our expectation since `M + LM = 5 + 1 = 6`

`xtevent y_base x_r, pol(z) panelvar(id) timevar(t) pre(0) post(5) overidpre(5) overidpost(1) cluster(id) norm(-1)`
`xteventplot`

![image](https://user-images.githubusercontent.com/70981003/177580957-a17d8e1c-2d10-4ed0-9bb6-eb1ef6359f3d.png)

**Comment by jmshapir** [comment]
Got it, thanks @nateschor!

When you have a chance, can you let me know what you think of my reasoning in https://github.com/JMSLab/xtevent/issues/13#issuecomment-1176313628?

**Comment by nateschor** [comment]
@jmshapir what I wanted to flag was that in the screenshot below, the farthest lag was given by `M + LM + 1` rather than `M + LM` (Veli was able to reproduce the behavior).

![image](https://user-images.githubusercontent.com/70981003/177582323-2e5eaf23-e3df-475a-b6cd-cda3bd86afdb.png)

![image](https://user-images.githubusercontent.com/70981003/177583068-85117fd0-b4b7-41c0-9027-33a92b551649.png)

------------------------------------------------------------

However, when I rerun the original code `xtevent y_base x_r, pol(z) panel(id) timevar(t) pre(2) post(3) overidpre(4) overidpost(5) norm(-3) plot` that yesterday produced the `3 + 5 + 1 = 9` lags, I am now only getting `3 + 5 = 8` lags (which is what we are expecting). Seems like @jorpppp addressed it in https://github.com/JMSLab/xtevent/issues/13#issuecomment-1176317658, thanks! 

![image](https://user-images.githubusercontent.com/70981003/177584186-43741093-4a84-4f3e-9d99-5246b88d7ac3.png)

**Comment by nateschor** [comment]
> Got it, thanks @nateschor!
> 
> When you have a chance, can you let me know what you think of my reasoning in [#13 (comment)](https://github.com/JMSLab/xtevent/issues/13#issuecomment-1176313628)?

@jmshapir I agree with your reasoning! 

> Thanks @nateschor! My understanding is that you're now seeing expected behavior from xtevent in the examples you were considering. If that's true then I agree we're good here and can leave this issue https://github.com/JMSLab/xtevent/issues/13 as resolved!

That's correct, I'm now seeing the expected behavior from `xtevent` and leaving https://github.com/JMSLab/xtevent/issues/13 as resolved sounds good to me, thanks!
